workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs

```diff
@@ -54,33 +55,68 @@ protected override void OnDestroyManager()
 
         protected override void OnUpdate()
         {
-            var updateSystem = World.GetExistingManager<ComponentUpdateSystem>();
             if (worker.Connection == null)
             {
                 return;
             }
 
-            do
+            try
             {
-                using (var opList = worker.Connection.GetOpList(0))
+                do
                 {
-                    try
-                    {
-                        Dispatcher.Process(opList);
-                    }
-                    catch (Exception e)
+                    using (var opList = worker.Connection.GetOpList(0))
                     {
-                        worker.LogDispatcher.HandleLog(LogType.Exception, new LogEvent("Exception:")
-                            .WithException(e));
+                        inCriticalSection = opDeserializer.ParseOpListIntoDiff(opList, diff);
                     }
                 }
+                while (inCriticalSection);
+
+                inCriticalSection = false;
```

Comment:
This will always already be false here

Reply:
Relic of previous way of doing it. Thought I'd deleted that.